File: exo5s1.m
x = -10:10;              
U = (x >= 0);           
V = exp(-0.3*abs(x));   

Y_add  = U + V;          % Addition
Y_mult = 0.6*U + 0.4*V;  
Y_trunc = U .* (abs(x)<=5); % Troncature
Y_rev = fliplr(U);       % Reverse

figure;  
subplot(2,2,1)
stem(x, Y_add)
title('Addition')

subplot(2,2,2)
stem(x, Y_mult)
title('Multiplication')

subplot(2,2,3)
stem(x, Y_trunc)
title('Troncature')

subplot(2,2,4)
stem(x, Y_rev)
title('Reverse')

File: exo5s1.py
import numpy as np
import matplotlib.pyplot as plt

x = np.arange(-10, 11)

# Signaux
U = (x >= 0).astype(int)          # signal1
V = np.exp(-0.3 * np.abs(x))      # signal 2

# Manipulations
Y_add = U + V                      
Y_mult = 0.6*U + 0.4*V             
Y_trunc = U * (np.abs(x) <= 5)     
Y_rev = U[::-1]                     


plt.figure()
plt.stem(x, Y_add, basefmt=" ")
plt.title('Add')

plt.figure()
plt.stem(x, Y_mult, basefmt=" ")
plt.title('mult')

plt.figure()
plt.stem(x, Y_trunc, basefmt=" ")
plt.title('trunk')

plt.figure()
plt.stem(x, Y_rev, basefmt=" ")
plt.title('Reverse')

plt.show()
